surveyor-brick/src/Surveyor/Brick.hs

```diff
@@ -312,18 +312,90 @@ stateFromContext :: forall arch s p sym ext rtp f a st fs
                  -> (PN.NonceGenerator IO s -> PN.Nonce s arch -> CFH.HandleAllocator -> IO (C.AnalysisResult arch s))
                  -> C.Chan (C.Events s (C.S BH.BrickUIExtension BH.BrickUIState))
                  -> LCSET.SimState p sym ext rtp f a
-                 -> C.Breakpoint sym
+                 -> Maybe (C.Breakpoint sym)
                  -> IO (C.S BH.BrickUIExtension BH.BrickUIState arch s)
 stateFromContext ng mkAnalysisResult chan simState bp = do
   let topFrame = simState ^. LCSET.stateTree . LCSET.actFrame
+  let simCtx = simState ^. LCSET.stateContext
+  let halloc = simCtx ^. L.to LCSET.simHandleAllocator
+  let addrParser _s = Nothing
   case topFrame ^. LCSET.gpValue of
     LCSC.RF {} -> error "Unexpected return value"
-    LCSC.OF {} -> error "Unexpected override frame"
+    LCSC.OF LCSC.OverrideFrame {} -> do
+      n0 <- PN.freshNonce ng
+      ares <- mkAnalysisResult ng n0 halloc
+      symCfg <- C.defaultSymbolicExecutionConfig ng
+      let sesID = symCfg ^. C.sessionID
+      case LCSET.activeFrames (simState ^. LCSET.stateTree) of
+        _ : LCSET.SomeFrame (LCSC.MF LCSC.CallFrame { LCSC._frameCFG = pfcfg }) : _ -> do
```

Comment:
Can we make this more general and find the first call frame in the list of active frames, instead of just assuming that the next frame is the call frame? I think that will be true in practice, but it is possible to call one override from another.

Reply:
I refactored this into a helper which walks the whole list.